src/geckomat/gather_kcats/getStandardKcat.m

```diff
@@ -0,0 +1,152 @@
+function [model, rxnsMissingGPR, standardMW, standardKcat] = getStandardKcat(model, modelAdapter, threshold)
+% getStandardKcat
+%   Calculate an standard kcat to rxns which a GPR have not been found in the
+%   model
+%
+% Input:
+%   model           an ecModel in GECKO 3 version
+%   modelAdapter    a loaded model adapter (Optional, will otherwise use the
+%                   default model adapter).
+%   threshold       a threshold to determine when use a kcat value based on
+%                   the mean kcat of the reactions in the same subSystem or
+%                   based on the median value of all the kcat in the model.
+%                   Second option is used when the number of reactions in a
+%                   determined subSystem is < threshold. (Optional, default = 10)
+%
+% Output:
+%   model           ecModel where model.ec is expanded with a standard
+%                   protein with standard kcat and standard MW, assigned to
+%                   reactions without gene associations.
+%   rxnsMissingGPR  a list of update rxns index with a standard value
+%   standardMW      the standard MW value calculated
+%   standardKcat    the standard Kcat value calculated
+%
+%   A pseudometabolite named 'prot_standard' will be added as well as a gene
+%   and enzyme named 'standard'. While model.ec.kcat is populated,
+%   applyKcatConstraints would still need to be run to apply the new
+%   constraints to the S-matrix.
+%
+% Usage:
+%    [model, rxnsMissingGPR, standardMW, standardKcat] = getStandardKcat(model, modelAdapter, threshold);
+
+if nargin < 2 || isempty(modelAdapter)
+    modelAdapter = ModelAdapterManager.getDefaultAdapter();
+    if isempty(modelAdapter)
+        error('Either send in a modelAdapter or set the default model adapter in the ModelAdapterManager.')
+    end
+end
+
+if nargin < 3
+    threshold = 10;
+end
+
+% Maybe this can be an input ???
+databases = loadDatabases('uniprot', modelAdapter);
+
+% An stardard MW is defined for all the rxns which does not have a GPR
+% rule defined. This is based in all the proteins reported for the specific
+% organism in uniprot
+standardMW = median(databases.uniprot.MW, 'omitnan');
+
+% An standard Kcat is defined for all the rxns which does not have a GPR
+% rule defined. In this case, the kcat value for a particular reaction is
+% defined as the mean of the kcat values of the reactions involved in the
+% same subsystem in which the given reaction is involved. Nevertheless, if
+% a subSystem have a number of reactions lower than a treshold, the kcat
+% value will be the median of the kcat in all the reactions of the model.
+
+% Get the kcat value based on all the kcats in the model
+standardKcat = median(model.ec.kcat, 'omitnan');
```

Comment:
I'm not sure how common zeros are in the kcat list, but since you check for them below, perhaps they should be omitted in this median calculation as well.

Reply:
They should indeed be ignored here, no matter how (un)common they are.

Reply:
few zeros, but I will include this